pysidt/sidt.py

```diff
@@ -932,6 +946,529 @@ def regularize(self, data=None, check_data=True):
+            return
+
+        ext,new_rule,comp_rule = self.choose_extension(parent, extlist)
+        
+        assert parent.name != "Root" or ext
+        
+        parent.items = total_items #fix parent.items now that we've picked an extension
+        
+        if ext is None:
+            logging.info(f"no extension selected skipping node {parent.name}")
+            self.skip_nodes.append(parent.name)
+            return
+        
+        new, comp = split_mols(parent.items, ext)
+        ind = extlist.index(ext)
+        grp, grpc, name, typ, indc = exts[ind]
+        
+        node = Node(
+            group=grp,
+            items=new,
+            rule=new_rule,
+            parent=parent,
+            children=[],
+            name=name,
+            depth=parent.depth + 1,
+        )
+
+        assert not (name in self.nodes.keys()), name
+
+        self.nodes[name] = node
+        parent.children.append(node)
+        self.new_nodes.append(name)
+
+        for k, datum in enumerate(self.datums):
+            for i, d in enumerate(self.mol_node_maps[datum]["mols"]):
+                if any(d is x for x in new):
+                    assert d.is_subgraph_isomorphic(
+                        node.group, generate_initial_map=True, save_order=True
+                    )
+                    self.mol_node_maps[datum]["nodes"][i] = node
+
+        logging.info("adding node {}".format(name))
+        
+        if grpc:
+            class_true = 0
+            frags = name.split("_")
+            frags[-1] = "N-" + frags[-1]
+            cextname = ""
+            for k in frags:
+                cextname += k
+                cextname += "_"
+            cextname = cextname[:-1]
+            nodec = Node(
+                group=grpc,
+                items=comp,
+                rule=comp_rule,
+                parent=parent,
+                children=[],
+                name=cextname,
+                depth=parent.depth + 1,
+            )
+
+            self.nodes[cextname] = nodec
+            parent.children.append(nodec)
+            self.new_nodes.append(cextname)
+
+            for k, datum in enumerate(self.datums):
+                for i, d in enumerate(self.mol_node_maps[datum]["mols"]):
+                    if any(d is x for x in comp):
+                        assert d.is_subgraph_isomorphic(nodec.group, generate_initial_map=True, save_order=True), (d.to_adjacency_list(),nodec.group.to_adjacency_list())
+                        self.mol_node_maps[datum]["nodes"][i] = nodec
+
+            parent.items = []
+        else:
+            parent.items = comp
+            parent.rule = comp_rule
+
+    def evaluate(self, mol):
+        """
+        Evaluate tree for a given possibly labeled mol
+        """
+        out = 0.0
+        decomp = self.decomposition(mol)
+        for d in decomp:
+            children = self.root.children
+            node = self.root
+            boo = True
```

Comment:
Rename all boo to not_done to improve clarity
```suggestion
            not_done = True
```